Add rmb() definition for AArch64 architecture.

definition copied from kernel source (like with rest of archs)

Upstream-status: pending
---
 libperf_events/operf_utils.h |    5 +++++
 1 file changed, 5 insertions(+)

--- oprofile-0.9.8.orig/libperf_events/operf_utils.h
+++ oprofile-0.9.8/libperf_events/operf_utils.h
@@ -146,10 +146,15 @@ void op_release_resources(void);
  */
 #define rmb()		((void(*)(void))0xffff0fa0)()
 #define cpu_relax()	asm volatile("":::"memory")
 #endif
 
+#ifdef __aarch64__
+#define rmb()		asm volatile("dmb ld" ::: "memory")
+#define cpu_relax()	asm volatile("yield" ::: "memory")
+#endif
+
 #ifdef __mips__
 #include <asm/unistd.h>
 #define rmb()		asm volatile(					\
 				".set	mips2\n\t"			\
 				"sync\n\t"				\
